app/src/main/java/com/example/moviedetails/network/MovieApi.kt

```diff
@@ -0,0 +1,19 @@
+package com.example.moviedetails.network
+
+import com.example.moviedetails.API_KEY
+import retrofit2.http.GET
+import retrofit2.http.Path
+import retrofit2.http.Query
+
+interface MovieApi {
+
+    @GET("movie/popular?")
+    suspend fun getMovies(@Query("api_key") api_key:String=API_KEY): APIResponse
```

Comment:
I can understand this. But we can add this API key by using `Interceptors` on OkHttp.  even something like bearer token we will try to add it at that level.